platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/service/WritableRosterStoreTest.java

```diff
@@ -0,0 +1,208 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.hedera.hapi.node.base.ServiceEndpoint;
+import com.hedera.hapi.node.state.primitives.ProtoBytes;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.hapi.node.state.roster.RosterState;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.RosterStateId;
+import com.swirlds.platform.state.MerkleRoot;
+import com.swirlds.platform.state.PlatformStateAccessor;
+import com.swirlds.platform.state.RosterStateAccessor;
+import com.swirlds.platform.state.RosterStateModifier;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import com.swirlds.platform.state.signed.SignedState;
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.state.merkle.singleton.SingletonNode;
+import com.swirlds.state.merkle.singleton.WritableSingletonStateImpl;
+import com.swirlds.state.spi.WritableKVState;
+import com.swirlds.state.spi.WritableStates;
+import com.swirlds.state.test.fixtures.MapWritableKVState;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class WritableRosterStoreTest {
+
+    private final WritableStates writableStates = mock(WritableStates.class);
+    private RosterStateModifier rosterStateModifier;
+    private RosterStateAccessor rosterStateAccessor;
+
+    private final SoftwareVersion version = mock(SoftwareVersion.class);
+    private final ReservedSignedState initialState = mock(ReservedSignedState.class);
+
+    @BeforeEach
+    void setUp() {
+        final SingletonNode<RosterState> rosterStateSingleton = new SingletonNode<>(
+                PlatformStateService.NAME,
+                RosterStateId.ROSTER_STATES_KEY,
+                0,
+                RosterState.PROTOBUF,
+                new RosterState(null, new LinkedList<>()));
+        final WritableKVState<ProtoBytes, Roster> rosters = MapWritableKVState.<ProtoBytes, Roster>builder(
+                        RosterStateId.ROSTER_KEY)
+                .build();
+        when(writableStates.<ProtoBytes, Roster>get(RosterStateId.ROSTER_KEY)).thenReturn(rosters);
+        when(writableStates.<RosterState>getSingleton(RosterStateId.ROSTER_STATES_KEY))
+                .thenReturn(new WritableSingletonStateImpl<>(RosterStateId.ROSTER_STATES_KEY, rosterStateSingleton));
+
+        rosterStateAccessor = new ReadableRosterStore(writableStates);
+        rosterStateModifier = new WritableRosterStore(writableStates);
+    }
+
+    @Test
+    @DisplayName("Test determine active roster without software upgrade")
```

Comment:
The display name on this test says it is determining active roster without software upgrade.  But the logic of the test doesn't access the active roster from the state.   I'm not sure how the test implements the description in the display name.

Reply:
fixed